lib/msal-browser/docs/caching.md

```diff
@@ -1,3 +1,70 @@
-## Cache Storage
+# Caching in MSAL
 
-We offer two methods of storage for Msal, `localStorage` and `sessionStorage`.  Our recommendation is to use `sessionStorage` because it is more secure in storing tokens that are acquired by your users, but `localStorage` will give you Single Sign On accross tabs and user sessions.  We encourage you to explore the options and make the best decision for your application.
+When MSAL acquires a token, it caches it for future usage. MSAL manages token lifetimes and refreshing for you. The `acquireTokenSilent()` API retrieves access tokens from the cache for a given account and renews them if needed.
+
+## Cache storage
+
+You can configure the cache storage location via the configuration object that is used to instantiate MSAL:
+
+```typescript
+import { PublicClientApplication, BrowserCacheLocation } from "@azure/msal-browser";
+
+const pca = new PublicClientApplication({
+    auth: {
+        clientId: "Enter_the_Application_Id_Here", // e.g. "b1b60dca-c49d-496e-9851-xxxxxxxxxxxx" (guid)
+        authority: "https://login.microsoftonline.com/Enter_the_Tenant_Info_Here", // e.g. "common" or your tenantId (guid),
+        redirectUri: "/"
+    },
+    cache: {
+       cacheLocation: BrowserCacheLocation.SessionStorage // "sessionStorage"
+    }
+});
+```
+
+By default, MSAL stores the various authentication artifacts it obtains from the IdP in browser storage using the [Web Storage API](https://developer.mozilla.org/docs/Web/API/Web_Storage_API) supported by all modern browsers. Accordingly, MSAL offers two methods of persistent storage: `sessionStorage` (default) and `localStorage`. In addition, MSAL provides `memoryStorage` option which allows you to opt-out of storing the cache in browser storage.
+
+| Cache Location   | Cleared on              | Shared between windows/tabs | Redirect flow supported |
+|------------------|-------------------------|-----------------------------|-------------------------|
+| `sessionStorage` | window/tab close        | No                          | Yes                     |
+| `localStorage`   | user logout             | Yes                         | Yes                     |
+| `memoryStorage`  | page refresh/navigation | No                          | No                      |
+
+> :bulb: While the authentication state may be lost in session and memory storage due to window/tab close or page refresh/navigation, respectively, users will still have an active session with the IdP as long as the session cookie is not expired and might be able to re-authenticate without any prompts.
+
+### Cookie storage
+
+MSAL Browser can be configured to use cookies for storing temporary authentication artifacts. This is useful for apps that need to support older browsers such as Internet Explorer. Note that when this option is chosen, tokens themselves are still stored in browser or memory storage. Please refer to [configuration](./configuration.md#cache-config-options) for more.
```

Comment:
Isn't internet explorer officially end of life at this point?  What still supported browsers **require** this?  If we can suggest folks not use cookies that would be my preference.

Reply:
I would rephrase this to specifically note that it is needed for browsers that may clear local/session storage during redirect-based login, which includes IE and Firefox Inprivate.

Reply:
(as opposed to saying this is needed only for older browsers)

Reply:
revised!